File: sessions/week-00-introduction/algorithms/javascript/find_occur.js
function strStr(haystack, needle) {
    if (needle === "") return 0;

    for (let i = 0; i <= haystack.length - needle.length; i++) {
        let match = true;
        for (let j = 0; j < needle.length; j++) {
            if (haystack[i + j] !== needle[j]) {
                match = false;
                break;
            }
        }
        if (match) return i;
    }

    return -1;
}

// Example:
console.log(strStr("leetcode", "code")); // Output: 4

File: sessions/week-00-introduction/algorithms/python/find_occur.py
def strStr(haystack, needle):
    if needle == "":
        return 0

    for i in range(len(haystack) - len(needle) + 1):
        match = True
        for j in range(len(needle)):
            if haystack[i + j] != needle[j]:
                match = False
                break
        if match:
            return i
    return -1

# Example:
print(strStr("leetcode", "code"))  # Output: 4
